### PR TITLE
I54 create subteam endpoint

### LINKED ISSUE #54: Create subteam endpoint
# Prerequisite
#40 Team API endpoint

# Information
Create a new API endpoint for the "Subteam" db table using the Django REST framework. This task is split into multiple small parts that each team member can tackle! It should be under the folder server/api/subteam

1. Create Subteam Model in `models.py`
2. Create Views in `views.py`
It should have views for the following operations:
- create a new subteam
- read/get details of subteam
- update details of the subteam
- delete subteam, be careful with this we dont want to lose existing data if the subteam id is a FK in another table
3.  Add url patterns to `urls.py`
4.  Register the app for the admin

You will be required to research how to complete this task!

TIP: You can use postman to test your endpoints!

Model data:
- subteam_id
- name
- team_id

### PR DESCRIPTION
**Pull request by YunnTun**
## Change Summary
**[Briefly summarise the changes that you made. Just high-level stuff]**

### Change Form
**Fill this up (NA if not available). If a certain criteria is not met, can you please give a reason.**

- [ ] The pull request title has an issue number
- [ ] The change works by "Smoke testing" or quick testing
- [ ] The change has tests
- [ ] The change has documentation

## Other Information
**[Is there anything in particular in the review that I should be aware of?]**

# Related Issue

- Resolve #54

### CODE REVIEW

File: server/api/subteam/models.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from django.db import models
+
+
+class SubTeam(models.Model):
+    subteam_id = models.AutoField(primary_key=True)
+    name = models.CharField(max_length=254)
+    # to add in later like chantelle said in issue
+    # team_id = models.ForeignKey()
```
Comment (ChantelleKerr) [inline server/api/subteam/models.py:8]: Team is in main now so you can make team_id a field :) You'll have to import the team model into its file

Comment (YunnTun) [reply server/api/subteam/models.py]: yup added

File: server/api/subteam/models.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from django.db import models
+from api.team.models import Team
+
+
+class SubTeam(models.Model):
+    subteam_id = models.AutoField(primary_key=True)
+    name = models.CharField(max_length=254)
+    team_id = models.ForeignKey(Team, on_delete=models.CASCADE, related_name='teamOf', default="")
```
Comment (ChantelleKerr) [inline server/api/subteam/models.py:8]: you dont need related_name or default

**Review by ChantelleKerr — CHANGES_REQUESTED** [review]
Remove some thing that we don't need :)

File: server/api/subteam/models.py (hunk 2 above)
Comment (YunnTun) [reply server/api/subteam/models.py:8]: yup have deleted unnecessary arguments and migrated

File: server/api/subteam/urls.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+from django.urls import path
+from . import views
+
+app_name = "subteam"
+urlpatterns = [
+    path("create_subteam/", views.create_subteam, name="create-subteam"),
+    path("get_subteams/", views.get_subteams, name="get-subteams"),
+    path("get_subteam/<int:subteam_id>", views.get_subteam, name="get-subteam"),
```
[the remark this answers is not in the document]
Comment (YunnTun) [reply server/api/subteam/urls.py:8]: fixed, hopefully its correct now (both here in urls and views)

**Review by ChantelleKerr — APPROVED** [review]
Great job! Approved

**Pull request merged**